verifiers/internal/gha/provenance.go

```diff
@@ -341,11 +341,20 @@ func getAsString(environment map[string]interface{}, field string) (string, erro
 
 	i, ok := value.(string)
 	if !ok {
-		return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "environment type string")
+		return "", fmt.Errorf("%w: %s '%s'", serrors.ErrorInvalidDssePayload, "environment type string", field)
 	}
 	return i, nil
 }
 
+func getAsAny(environment map[string]interface{}, field string) (interface{}, error) {
```

Comment:
Might as well actually make it 'any`

```suggestion
func getAsAny(environment map[string]any, field string) (any, error) {
```